Google.Api.Gax.Grpc/Rest/RestMethod.cs

```diff
@@ -115,8 +116,7 @@ internal static RestMethod Create(MethodDescriptor method, JsonParser parser)
                 var path = pathFactory(message);
                 var body = bodyFactory(message);
 
-                var fieldsToEncode = unusedEligibleFields.Where(field =>
-                    field.IsRequired || !IsDefaultValueForTranscoding(field, field.Accessor.GetValue(message)));
+                var fieldsToEncode = unusedEligibleFields.Where(field => !field.HasPresence || field.HasPresence && field.Accessor.HasValue(message));
```

Comment:
I would *definitely* use extra brackets here to make the precedence clear. I assume you want `!field.HasPresence || (field.HasPresence && field.Accessor.HasValue(message))`.

Alternatively, given that `||` is short-circuiting, just skip the second HasPresence check: `!field.HasPresence || field.Accessor.HasValue(message)`.

Reply:
👍